XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,403 @@
+
+- The Oracle space key (0x52)
+- The Owner Account ID, `Owner`.
+- The Oracle Document ID, `OracleDocumentID`. This field describes a unique Price Oracle instance for the given account. The Oracle Document ID is maintained by the Oracle Provider.
+
+The `Owner` and `OracleDocumentID` uniquely identify the `PriceOracle` object and must be passed to the Oracle transactions.
+
+#### Example of `PriceOracle` JSON
+
+    {
+        "LedgerEntryType": "PriceOracle",
+        "Owner": "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW",
+        "Provider": "70726F7669646572",
+        "AssetClass": "63757272656E6379",
+        "PriceDataSeries": [
+          {
+            "PriceData": {
+              "BaseAsset": "XRP",
+              "QuoteAsset": "USD",
+              "AssetPrice": 74,
+              "Scale": 2,
+            }
+          },
+        ],
+        "LastUpdateTime": 743609414,
+        "PreviousTxnID": "C53ECF838647FA5A4C780377025FEC7999AB4182590510CA461444B207AB74A9",
+        "PreviousTxnLgrSeq": 56865244
+    }
+
+## Transactions
+
+This proposal introduces several new transactions to allow for the creation, update, and deletion of the `PriceOracle` object.
+
+### Transaction for creating or updating `PriceOracle` instance
+
+We define a new transaction **OracleSet** for creating or updating a `PriceOracle` instance.  Before the transaction can be submitted to create a new `PriceOracle` instance, the Oracle Provider has to do the following:
+
+- Create or own the `Owner` on the XRPL with sufficient XRP balance to meet the XRP reserve and the transaction fee requirements.
+- The Oracle Provider has to publish the `Owner` account public key so that it can be used for verification by dApp’s.
+- The Oracle Provider has to publish a registry of available Price Oracles with their unique `OracleDocumentID`. The hash of the `Owner` and the `OracleDocumentID` uniquely identifies the Price Oracle on-ledger object.
+
+#### Example of OracleSet transaction JSON
+
+    {
+        "TransactionType": "OracleSet",
+        "Account": "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW",
+        "OracleDocumentID": 34,
+        "Provider": "70726F7669646572",
+        "LastUpdateTime": 743609014,
+        "AssetClass": "63757272656E6379",
+        "PriceDataSeries": [
+          {
+            "PriceData": {
+              "BaseAsset": "XRP",
+              "QuoteAsset": "USD",
+              "AssetPrice": 740,
+              "Scale": 3
+            }
+          }
+        ]
+    }
+
+#### Transaction fields for **OracleSet** transaction
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+| `TransactionType` | :heavy_check_mark: | `string`| UINT16 |
+| `Account` | :heavy_check_mark: | `string` | `ACCOUNTID` |
+| `OracleDocumentID` | :heavy_check_mark: | `string` | `UINT32` |
+| `Provider` | :grey_question: | `string` | `BLOB` |
```

Comment:
The question mark here and in `AssetClass` make it seem like the spec is unfinished - IMO it should just be labeled as "not required" and further explained below.

Reply:
It can always be changed later if we realize it's confusing, so it's not too big of a deal.

Reply:
Yeah, I don't really care what mark is used, I just wanted something besides "no mark", which I (and I think other readers) interpret as "never required".